Add optional GroupBy() 'aggregate' field.

benbjohnson opened the pull request:
## Overview

This commit adds an `aggregate` field that allows a `Sum()` call
to be executed for every returned group.

Fixes #1825

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

benbjohnson commented:
@jaffee I made the `aggregate` changes as we discussed. Can you look over this and verify that I'm doing it right? I need to add some additional test coverage and checks but I wanted to make sure it's in the right direction.

jaffee commented on executor.go:
```diff
@@ -566,12 +566,11 @@ func (e *executor) executeBitmapCallShard(ctx context.Context, index string, c *
 
-	var filter *Row
-	if len(c.Children) == 1 {
+	if filter == nil && len(c.Children) == 1 {
```
a comment explaining the difference between the children and the explicit filter arg would not go awry.

jaffee commented:
@benbjohnson looks good - this jives with my understanding

benbjohnson commented:
@jaffee Sorry for the long delay on this. I added a blurb in `docs/query-language.md` describing the change and I added the comment requested above. They're both pretty minor changes so I rebased and force pushed.

jaffee commented:
as I discovered just today... I think we need to explicitly do translation on the aggregate Call. See #1970 

Also a test for that would not go awry.

jaffee commented:
probably better if you review #1970 first, we merge it, then you can rebase on it and add the test to the list of test cases in that PR. Should be less work all around.
